### PR TITLE
New League of Legends Cheat Sheet

### PR DESCRIPTION
**Pull request by CedArctic**
## Description of new Instant Answer, or changes

This Instant Answer responds to user's queries such as "lol cheat sheet" or "lol commands" with a cheat sheet containing all the Hotkeys and Chat Commands available in the popular MOBA game, League of Legends.

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/league_of_legends_hotkeys_and_commands

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by pjhampton** [comment]
Hey @CedArctic. Welcome 👋 

Thanks for your contribution. Unfortunately, this doesn't meet our standards for a PR / Issue. Please see our [Contributing Guide](https://duckduckhack.com/contributing.html). Please make sure that the PR template is filled out and ping me to reopen.

**Comment by CedArctic** [comment]
@pjhampton I updated my first comment with a description of the IA. If there is anything else needed please tell me :) .

**Comment by pjhampton** [comment]
Thanks @CedArctic 👍

File: share/goodie/cheat_sheets/json/lol.json [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    "id": "lol_cheat_sheet",
+    "name": "League of Legends Hotkeys And Commands",
+    "description": " This is a cheat sheet for basic Hotkeys and Commands for the popular MOBA game League of Legends (LoL). ",
+    "metadata": {
+        "sourceName": "League of Legends Wiki: Hotkeys and commands",
+        "sourceUrl": "http://leagueoflegends.wikia.com/wiki/Hotkeys_and_commands"
+    },
+    "aliases": [
+        "league of legends"
+    ],
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: I think you should add some more `aliases`

File: share/goodie/cheat_sheets/json/lol.json [hunk 2]
```diff
@@ -0,0 +1,226 @@
+{
+    "id": "lol_cheat_sheet",
+    "name": "League of Legends Hotkeys And Commands",
+    "description": " This is a cheat sheet for basic Hotkeys and Commands for the popular MOBA game League of Legends (LoL). ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: Remove that extra space from the end

File: share/goodie/cheat_sheets/json/lol.json [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+            "val": "Add an ability point to the champion's 2nd ability.",
+            "key": "[Ctrl] + [W]"
+        }, {
+            "val": "Add an ability point to the champion's 3rd ability.",
+            "key": "[Ctrl] + [E]"
+        }, {
+            "val": "Add an ability point to the champion's ultimate ability.",
+            "key": "[Ctrl]+[R]"
+        }, {
+            "val": "Cast champion's 1st ability on self (must be a support spell). ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: Remove extra space from the end

File: share/goodie/cheat_sheets/json/lol.json [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+            "val": "Cast 1st Summoner spell.",
+            "key": "D"
+        }, {
+            "val": "Cast 2nd Summoner spell.",
+            "key": "F"
+        }, {
+            "val": "Use consumable item in corresponding inventory slot.",
+            "key": "Numbers [1-3] and [5-7]"
+        }, {
+            "val": "Use designated active ability items. ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: Remove extra space form the end

File: share/goodie/cheat_sheets/json/lol.json [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+            "val": "Center camera on champion (holding Space will hold the camera on the champion until released).",
+            "key": "Space"
+        }, {
+            "val": "Open match stats page/leaderboard.",
+            "key": "O"
+        }, {
+            "val": "Open match stats page/leaderboard (unlike O, the screen will disappear when the Tab button is released).",
+            "key": "Tab"
+        }, {
+            "val": "Toggle retreat ping cursor. Clicking with this hotkey on an allied champion (or their icon in the upper left area of the screen) will show an alert telling that specific champion to fall back.",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: Too long sentence, make it a little short.

File: share/goodie/cheat_sheets/json/lol.json [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+            "val": "Toggles the effect for your Champion if applicable.",
+            "key": "[Ctrl] + [5]"
+        }, {
+            "val": "Displays your Mastery Emote.",
+            "key": "[Ctrl] + [6]"
+        }, {
+            "val": "Pauses the game (Only works in certain situations, like tournaments).",
+            "key": "/pause, /stop, /s"
+        }, {
+            "val": "Adds text to a text file called 'MyNotes.txt'",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: If you are using `period` to end the sentence then let's be consistent

File: share/goodie/cheat_sheets/json/lol.json [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+            "key": "/note, /n"
+        }, {
+            "val": "Enables or disables a filter for profanity ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: Both `period` and extra space

File: share/goodie/cheat_sheets/json/lol.json [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+            "val": "Will mute all text from a player for the duration of your current game session.",
+            "key": "/mute 'summoner name'"
+        }, {
+            "val": "Will mute a player during this game session and all subsequent game sessions.",
+            "key": "/mute all"
+        }, {
+            "val": "Will mute all players during this game session and all subsequent game sessions.",
+            "key": "/ignore 'summoner name'"
+        }, {
+            "val": "Go To Next Idle Villager",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: same here `period`

File: share/goodie/cheat_sheets/json/lol.json [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+            "key": "/ignore all"
+        }]
+
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: remove this extra line

**Comment by CedArctic** [comment]
Thanks for the suggestions @amarlearning , I've added the commits accordingly :) . However I think the terminal template doesn't look really well with the key bindings :/ .

File: share/goodie/cheat_sheets/json/lol.json [hunk 10]
```diff
@@ -0,0 +1,225 @@
+{
+    "id": "lol_cheat_sheet",
+    "name": "League of Legends Hotkeys And Commands",
+    "description": " This is a cheat sheet for basic Hotkeys and Commands for the popular MOBA game League of Legends (LoL).",
+    "metadata": {
+        "sourceName": "League of Legends Wiki: Hotkeys and commands",
+        "sourceUrl": "http://leagueoflegends.wikia.com/wiki/Hotkeys_and_commands"
+    },
+    "aliases": [
+        "league of legends", "league"
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: split this into separate lines

File: share/goodie/cheat_sheets/json/lol.json [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+    ],
+    "template_type": "terminal",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: You are right `terminal` doesn't look good. revert this change

File: share/goodie/cheat_sheets/json/lol.json [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+            "val": "Lists / commands and provides descriptions.",
+            "key": "/help, /?, /allcommands, /allcmds"
+        }, {
+            "val": "Starts vote for team surrender.",
+            "key": "/surrender, /ff, /concede, /forfeit"
+        }, {
+            "val": "Starts vote for a team with one or more afk (under certain conditions) to remake a game.",
+            "key": "/remake"
+        }, {
+            "val": "Plays Champion's joke",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: Period

File: share/goodie/cheat_sheets/json/lol.json [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+            "val": "Pauses the game (Only works in certain situations, like tournaments).",
+            "key": "/pause, /stop, /s"
+        }, {
+            "val": "Adds text to a text file called 'MyNotes.txt'.",
+            "key": "/note, /n"
+        }, {
+            "val": "Enables or disables a filter for profanity.",
+            "key": "/chatfilter"
+        }, {
+            "val": "Will send a message to another player. Summoner names are case sensitive.",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/lol.json]: This can be written as `Will send a message to another player (Summoner names are case sensitive).`

**Comment by CedArctic** [comment]
@amarlearning All your suggested changes have been made :)

**Comment by amarlearning** [comment]
Need to be reviewed by senior dev! :smile: 
cc @pjhampton @moollaza

**Comment by CedArctic** [comment]
@sahildua2305 I added the 4-space indentation :)

File: share/goodie/cheat_sheets/json/lol.json [hunk 14]
```diff
@@ -0,0 +1,296 @@
...
+                "val": "Open item shop.",
+                "key": "P"
+            },
+            {
+                "val": "Scroll camera in the corresponding direction.",
+                "key": "Arrow Keys"
+            },
+            {
+                "val": "Ping location/Open Radial Menu.",
+                "key": "[Alt]+[Left Click]"
```
Comment (TiagoDanin) [inline share/goodie/cheat_sheets/json/lol.json]: Two different styles (`[key]+[key]` and `[key] + [key]`)
I recommend using only one

**Comment by CedArctic** [comment]
@TiagoDanin Thanks for the tip! 😄

File: share/goodie/cheat_sheets/json/lol.json [hunk 15]
```diff
@@ -0,0 +1,296 @@
...
+    "id": "lol_cheat_sheet",
+    "name": "League of Legends Hotkeys And Commands",
+    "description": " This is a cheat sheet for basic Hotkeys and Commands for the popular MOBA game League of Legends (LoL).",
+    "metadata": {
+        "sourceName": "League of Legends Wiki: Hotkeys and commands",
+        "sourceUrl": "http://leagueoflegends.wikia.com/wiki/Hotkeys_and_commands"
+    },
+    "aliases": [
+        "league of legends",
+        "league"
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/lol.json]: @CedArctic In my opinion, `league` should not be there in the aliases.

**Comment by CedArctic** [comment]
@sahildua2305 I removed the alias you referred. Although I can't see any search query that it would conflict with, players of the game often refer to it as simply league as well. Also it was suggested earlier that the IA needs more aliases (without league it has just one).

**Comment by sahildua2305** [comment]
@CedArctic Ah! You have a valid point here. Considering that, we can add back the `league` alias. 👍

**Comment by CedArctic** [comment]
@sahildua2305 If at any point the alias creates a conflict we can remove it. But since DuckDuckGo requires the word league + (dictionary of cheat sheet words) to trigger the IA I don't think there should be a problem. :)

File: share/goodie/cheat_sheets/json/lol.json (hunk 15 above)
Comment (kirkins) [inline share/goodie/cheat_sheets/json/lol.json]: Yes lol can also mean laugh out loud but I feel like in the context of "lol cheat sheet" there is a 90% chance the person means league.

@sahildua2305 @CedArctic what do you think?

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/lol.json]: @kirkins good point. As you pointed out - "lol cheat sheet" in that context will mostly mean "league of legends". I don't have a strong feeling about this though. Can be convinced otherwise.

**Comment by sahildua2305** [comment]
@CedArctic yes, you're right.

**Comment by daxtheduck** [comment]
@CedArctic this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!
